backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutEngine.java

```diff
@@ -62,8 +63,10 @@ public IncrementalLayoutEngine(NodeSizeProvider nodeSizeProvider) {
         this.nodeSizeProvider = Objects.requireNonNull(nodeSizeProvider);
     }
 
-    public void layout(Optional<IDiagramEvent> optionalDiagramElementEvent, DiagramLayoutData diagram) {
+    public void layout(Optional<IDiagramEvent> optionalDiagramElementEvent, DiagramLayoutData diagram, LayoutConfigurator layoutConfigurator) {
         this.nodePositionProvider.reset();
+        this.nodeLabelPositionProvider.setLayoutConfigurator(layoutConfigurator);
+        this.edgeLabelPositionProvider.setLayoutConfigurator(layoutConfigurator);
```

Comment:
Do not use a setter, give those arguments to the constructor instead

Reply:
Done